migration-guide.adoc

```diff
@@ -37,4 +37,7 @@ To retain backwards compatibility, configure the setting `hibernate.type.preferr
 Previously, since Hibernate 6.0, columns of type `number` with scale 0 on Oracle were interpreted as `boolean`, `tinyint`, `smallint`, `int`, or `bigint`,
 depending on the precision.
 
-Now, columns of type `number` with scale 0 are interpreted as `int` or `bigint` depending on the precision.
\ No newline at end of file
+Now, columns of type `number` with scale 0 are interpreted as `int` or `bigint` depending on the precision.
+
+=== Merging stale objects
+An OptimisticLockException will now be thrown when a previously deleted instance of an entity that has optimistic locking enabled, is merged by another session. Note however that, at the moment, for this to work correctly with primitive version types (int, short and long) these have to be initialized to 0. For the primitive wrapper types this will work as-is.
```

Comment:
```suggestion
=== Merge versioned entity when row is deleted

So far, merging detached entities with a version assigned, resulted in an insert when the row was not found. Since this is intuitive and goes against the principle of optimistic locking, this was fixed to now throw an `OptimisticLockException` if the detached entity has a version assigned that is non-default value. For object type versions, any non-null value is considered non-default. For primitive type versions, any value other than the JVM default value (`0`) is considered non-default.
```

Reply:
Um, there's are some incorrect comma usage in both the original and the suggested change. You can't put a comma between the subject and its verb, I suppose. Also, did you mean "unintuitive"? Also, did you mean "**a** non-default value".

Reply:
I hate commas, so yeah, probably I misplaced one somewhere. Please suggest the correct version.

Reply:
Previously, merging a detached entity with an assigned version resulted in a SQL insert whenever there was no matching row in the database (for example, if the object had been deleted in another transaction). This behavior was unexpected and violated the rules of optimistic locking. 

An `OptimisticLockException` is now thrown in this situation if the version of the detached entity has any value other than its Java default value:

- For object types, `null` is the default value. 
- For primitive types, zero is the default value.